src/modules/baseCompare/baseCompare.js

```diff
@@ -0,0 +1,69 @@
+import GitHubAPI from 'github'
+import config from 'config'
+import ScreenshotBucket from 'server/models/ScreenshotBucket'
+
+export async function getBaseScreenshotBucket(options) {
+  const github = new GitHubAPI({ debug: config.get('env') === 'development' })
+
+  // http://stackoverflow.com/questions/9179828/github-api-retrieve-all-commits-for-all-branches-for-a-repo
+  // http://mikedeboer.github.io/node-github/#api-repos-getBranch
+  const baseCommits = await github.repos.getCommits({
+    owner: options.owner,
+    repo: options.repo,
+    sha: options.baseCommit,
+  })
+
+  const compareCommits = await github.repos.getCommits({
+    owner: options.owner,
+    repo: options.repo,
+    sha: options.compareCommit,
+  })
+
+  let forkCommit
+  const foundFork = compareCommits.data.some((compareCommit) => {
```

Comment:
You can reduce it to `forkCommit` instead of `foundFork` and `forkCommit` by switching `compareCommits` and `baseCommits` loops and using a find for the `baseCommits` one.